### ISSUE TITLE
Question: IAM Policy for cfncluster to work

### ISSUE THREAD

**Issue by adavanisanti** [aws/aws-parallelcluster#437]
Is there a specific IAM role that is needed for cfncluster to work? It seems to work only with "AdminAccess" Role. I have tried with "PowerUser" Role and it fails. It would be great if there is an IAM policy (json) that I can attach. Appreciate any help!

Thanks
Santi

**Comment by sean-smith** [comment]
Yep! See https://cfncluster.readthedocs.io/en/latest/iam.html

**Comment by adavanisanti** [comment]
Thanks!

**Comment by adavanisanti** [comment]
I have created a new user and attached CfnClusterUserPolicy according to instructions [here](https://cfncluster.readthedocs.io/en/latest/iam.html). I have also created CfnClusterInstancePolicy, created a `CfnClusterEAMRole` and attached the same to the user. 

I am using cfncluster version `1.5.4`. I get the following error `Config sanity error: You are not authorized to perform this operation`. Could you help me debug this?

**Comment by sean-smith** [comment]
Where did you attach the `CfnClusterUserPolicy`?

**Comment by adavanisanti** [comment]
I created a group, attached the policy to the group. I then added the user to the group.

**Comment by adavanisanti** [comment]
Sorry, `s3:GetObject` permissions were missing both on `User` and `Instance` policies. I don't see `authorization` error anymore. 

However, I see an error `AWS::AutoScaling::AutoScalingGroup ComputeFleet Received 2 FAILURE signal(s) out of 2. Unable to satisfy 100% MinSuccessfulInstances Percent Requirement`. 

I am trying to start a cluster with 2 `t2-medium` instances and a custom ami. Below is my config
```
[aws]
aws_region_name = <>
aws_access_key_id = <>
aws_secret_access_key = <>

[cluster default]
vpc_settings = <>
key_name = <>
master_instance_type = t2.medium
compute_instance_type = t2.medium
initial_queue_size = 2
max_queue_size = 2
maintain_initial_size = true
master_root_volume_size = 100
compute_root_volume_size = 100
base_os = ubuntu1604
ebs_settings = custom
scheduler = slurm
custom_ami = ami-<>
ec2_iam_role=`CfnClusterEC2Role`

[ebs custom]
volume_type = gp2
volume_size = 100

[vpc ]
master_subnet_id = <>
vpc_id = <>

[global]
update_check = true
sanity_check = true
cluster_template = default
```

**Comment by adavanisanti** [comment]
@sean-smith Looks like my custom ami is built using a base ami from version `1.4` and the cfncluster I'm using `1.5.4` resulting in errors.

I started a cluster with `--norollback` and looked the compute nodes `*cfn*log` files. Error is `ToolError: Command compute_ready failed`. 

I was able to start a cluster without `custom-ami`. 
I am assuming I need to rebuild an AMI for the latest cfncluster base ami.
